### PR TITLE
Add file_read_timer for json scanner

### PR DESCRIPTION
**Pull request by rickif**
## What type of PR is this：
- [ ] bug
- [ ] feature
- [x] enhancement
- [ ] others

## Which issues of this PR fixes ：

Fixes none.

## Problem Summary(Required) ：

This adds `FileReadTimer` for `JsonScanner` to provide facility for profile.

### CODE REVIEW

File: be/src/exec/vectorized/json_scanner.cpp [hunk 1]
```diff
@@ -617,10 +617,13 @@ Status JsonReader::_read_and_parse_json() {
...
     // TODO: Remove the down_cast, should not rely on the specific implementation.
     StreamLoadPipeInputStream* stream_file = down_cast<StreamLoadPipeInputStream*>(_file->stream().get());
-    // For efficiency reasons, simdjson requires a string with a few bytes (simdjson::SIMDJSON_PADDING) at the end.
-    RETURN_IF_ERROR(stream_file->read_one_message(&_json_binary_ptr, &length, simdjson::SIMDJSON_PADDING));
-    if (length == 0) {
-        return Status::EndOfFile("EOF of reading file");
+    {
+        SCOPED_RAW_TIMER(&_counter->file_read_ns);
+        // For efficiency reasons, simdjson requires a string with a few bytes (simdjson::SIMDJSON_PADDING) at the end.
+        RETURN_IF_ERROR(stream_file->read_one_message(&_json_binary_ptr, &length, simdjson::SIMDJSON_PADDING));
```
Comment (chaoyli) [inline be/src/exec/vectorized/json_scanner.cpp:623]: If the message arrives row by row, does it the counter will occupy much time?

Comment (rickif) [reply be/src/exec/vectorized/json_scanner.cpp:623]: https://gist.github.com/stunpix/54c304bc010307562ec3434dbd5e76ae
With this code, I have done a rough benchmark about the `clock_gettime(MONOTONIC)`, which is the implement of `SCOPED_RAW_TIMER`. The result is about `20ns/call`. It seems fast enough.

Comment (chaoyli) [reply be/src/exec/vectorized/json_scanner.cpp:623]: How many rounds do you run?

**Review by chaoyli — APPROVED** [review]

**Review by meegoo — APPROVED** [review]

**Review by decster — APPROVED** [review]

**Comment by rickif** [comment]
@mergifyio backport branch-2.2

**Comment by rickif** [comment]
@Mergifyio backport branch-2.1

**Comment by rickif** [comment]
@mergifyio backport branch-2.0

**Pull request merged**